src/main/java/edu/cornell/kfs/fp/dataaccess/impl/RecurringDisbursementVoucherSearchDaoJdbc.java

```diff
@@ -90,6 +89,16 @@ private String getSavedDvsSpawnedByRecurringDvForCurrentAndPastFiscalPeriodsSele
         return sqlBuilder.toString();
     }
 
+    @Override
+    public String findRecurringDvIdForSpawnedDv(String spawnedDvDocumentNumber) {
+        CuSqlQuery query = new CuSqlChunk()
+                .append("SELECT RECURRING_DV_DOC_NBR FROM KFS.FP_RCDV_DTL_T WHERE DV_DOC_NBR = ")
+                .appendAsParameter(spawnedDvDocumentNumber)
+                .toQuery();
```

Comment:
Are you referring to the `VER_NBR` column on our various tables? That's meant for using "optimistic" locks to prevent multiple users from overwriting each others' changes to a given entry; I don't see how that would help with this query. If you're referring to `RECURRING_DV_DOC_NBR` or `DV_DOC_NBR`, those contain document numbers, not version numbers.

As for duplication concerns, a given DV should not be spawnable by more than one Recurring DV, so I don't see a need to check for duplicates here. (There may be multiple table entries with a null `DV_DOC_NBR`, but in this case, we should not be searching for null IDs.)

Reply:
Okay, thanks for the explanation